### DISCUSSION TITLE
Is there a way to run FMSET non-interactively?

### DISCUSSION THREAD

**Discussion by vlbi-jun-yang** [nvi-inc/fs#156] [Q&A] [answered]
I am wondering whethere there is a way to run FMSET without some interactive inputs.  This will help me to quickly change the FILA10G configuration. At Onsala, this change is frequencly required to support different observations, e.g. e-VLBI, FRB observations with a different recorder. So, it would be great to run FMSET directly with the input options in the command line.

**Comment by wehimwich** [comment, marked as answer]
The short is answer is that currently there is no way to do this.

I am a little surprised that the situation occurs often enough and is laborious enough that a short-cut would be useful, but I hear you.

I expect that we can make something like this work. This may be particular easy if what you mean by _FiLa10g configuration_ is what comes from the _fila10g_cfg.ctl_ control file. My only concern about this is that there is other setup that is applied as part of the sync operation. Just applying the FiLa10G configuration would  mean that those things would not get done. I can imagine that for a given observing mode, somethings independent of the other setup, maybe the ethernet setup, could be changed independently. The user still would have to make sure that the full setup is applied first. We would be losing the current enforced requirement that it be set.

It is question for you users, whether it is enough of convenience to being able to do the FiLa10G configuration separately that it outweighs the risk of missing other parts of the setup. I would be happy to add this functional if it will improve operations, but I don't want to put out something that could cause problems and loss of data.

In more detail, when syncing, _fmset_ sends to the FiLa10G (except as noted), in order :

1. clock rate (to the DBBC)
2. sync (to the DBBC)
3. FiLa10G configuration
4. input_select
5. sample_rate
6. reset
7. the time

The only sticking point for doing this all non-interactively is that currently the operator is expected to verify the time (and possibly adjust it) interactively afterwards. Since that is interactive, setting the FiLa10G configuration the same way doesn't _seem_ like a big increase in the burden.

With some work, we could probably also verify the  time reliably non-interactively, but we haven't gone there yet. This is on the assumptions that: it is critical enough, infrequent enough, and that it might be complicated enough to make the non-interactive implementation handle all problems, that it is easier to do it interactively. We can of course revisit those assumptions; they may be outdated. I am a fan of automation, but I am not a regular user of the FiLa10G and am not sure I would see enough of the edges cases to make this reliable enough. Input, testing, and code from the community could help push that along.
 
If only the FiLa10G configuration needs to be sent, and isn't dependent on the other items, we could probably isolate that and only send it. Doing this through _fmset_ might be a bit overly complicated (but would allow applying other setup that it doesn't potentially affect the time). It might be better to have a script that can be run by a SNAP procedure to send the requested configuration from the _fila10g_cfg.ctl_ control file. That might be pretty easy to implement. 

Anyway, I think we may need more discussion about whether we want this change in general.

**Comment by vlbi-jun-yang** [comment]
Many thanks! Here, I only need change the FiLa10g configuration non-interactively. Most stations may not require this function. I know how to make a SNAP procedure to send the configuration.  While, I have got a hardware solution. The 2nd FiLa10G ethernet port is connected. This gives me the more flexibility now.   

In the near future, each EVN station is supposted to have two or more Flexbuffs to support 4 Gbps VLBI observations. It would be great to have a simple command to change recorders and FiLA10G configurations dynamically (without restarting FS).

**Comment by vlbi-jun-yang** [comment]
I have just realized that this Q&A should go to the webpage of Discussions since it is not an issue. Sorry!

**Comment by wehimwich** [comment]
No need to apologize of course. I guess it could be a discussion topic. I wasn't aware Discussions had been enabled. I wonder who might have done that (@dehorsley ?).  I will move it there for now. Some other existing issues might be better as discussions, but I am not sure we need to be too strict about that.

**Comment by wehimwich** [comment]
If there is more interest in this as a feature we can turn this back into an issue. I am still a little concerned about doing this as a SNAP procedure since this means you could need to maintain two different versions of the setups involved. if they don't change very often that is less of a burden, but it also means the issue will be out-of-mind when a change is needed. I think a script would be a better way to go and others may find it useful so it could be included in the FS.

How near is the future with two or more FlexBuffs and would they be used in parallel and/or serially?

**Reply by vlbi-jun-yang** [reply]
At the end of this year, each EVN station is supposed to get a second Flexbuff to double the available disk space. While, I am afraid that the goal might not be reached for various reasons at some stations. 

These Flexbuffs are supposed to be used serially for the normal VLBI observations with DBBC2. It is better to use a script to change the recorder. This script will help us to edit the control file mk5ad.ctl, run the command "fila10g=destination ..." and then restart FS.

**Reply by wehimwich** [reply]
I will open a new Discussion on controlling two FlexBuffs serially.
